metrics-operator/controllers/common/providers/prometheus/prometheus.go

```diff
@@ -67,3 +67,45 @@ func (r *KeptnPrometheusProvider) EvaluateQuery(ctx context.Context, metric metr
 	}
 	return value, b, nil
 }
+
+func EvaluateQueryWithRange(ctx context.Context, metric metricsapi.KeptnMetric, r *KeptnPrometheusProvider, api prometheus.API) (model.Value, prometheus.Warnings, error) {
```

Comment:
as both the `EvaluateQuery...` functions are only used here, i think we can make them unexported